x-pack/solutions/security/plugins/elastic_assistant/server/ai_assistant_service/index.ts

```diff
@@ -229,29 +245,124 @@ export class AIAssistantService {
-        const response = await createPipeline({
+        // update component template for semantic_text field
+        // rollover
+        let mappings: IndicesGetFieldMappingResponse = {};
+        try {
+          mappings = await esClient.indices.getFieldMapping({
+            index: '.kibana-elastic-ai-assistant-knowledge-base-default',
+            fields: ['semantic_text'],
+          });
+        } catch (error) {
+          /* empty */
+        }
+
+        const isUsingDedicatedInferenceEndpoint =
+          (
+            Object.values(mappings)[0]?.mappings?.semantic_text?.mapping?.semantic_text as {
+              inference_id: string;
+            }
+          )?.inference_id === ASSISTANT_ELSER_INFERENCE_ID;
+
+        if (knowledgeBaseDataStreamExists && isUsingDedicatedInferenceEndpoint) {
+          const currentDataStream = this.createDataStream({
+            resource: 'knowledgeBase',
+            kibanaVersion: this.options.kibanaVersion,
+            fieldMap: {
+              ...omit(knowledgeBaseFieldMap, 'semantic_text'),
+              semantic_text: {
+                type: 'semantic_text',
+                array: false,
+                required: false,
+                inference_id: ASSISTANT_ELSER_INFERENCE_ID,
+                search_inference_id: ELASTICSEARCH_ELSER_INFERENCE_ID,
+              },
+            },
+          });
+
+          // Add `search_inference_id` to the existing mappings
+          await currentDataStream.install({
+            esClient,
+            logger: this.options.logger,
+            pluginStop$: this.options.pluginStop$,
+          });
+
+          // Migrate data stream mapping to the default inference_id
+          const newDS = this.createDataStream({
+            resource: 'knowledgeBase',
+            kibanaVersion: this.options.kibanaVersion,
+            fieldMap: {
+              ...omit(knowledgeBaseFieldMap, 'semantic_text'),
+              semantic_text: {
+                type: 'semantic_text',
+                array: false,
+                required: false,
+              },
+            },
+            settings: {
+              // force new semantic_text field behavior
+              'index.mapping.semantic_text.use_legacy_format': false,
+            },
+            writeIndexOnly: true,
+          });
+
+          // We need to first install the templates and then rollover the indices
+          await newDS.installTemplates({
+            esClient,
+            logger: this.options.logger,
+            pluginStop$: this.options.pluginStop$,
+          });
+
+          const indexNames = // @ts-expect-error
+            (await esClient.indices.getDataStream({ name: newDS.name })).data_streams.map(
+              (ds) => ds.name
+            );
+
+          try {
+            await Promise.all(
+              indexNames.map((indexName) => esClient.indices.rollover({ alias: indexName }))
+            );
+          } catch (e) {
+            /* empty */
+          }
```

Comment:
It would be nice to integrate this logic inside the adapter, we could add a ` migration: 'update' | 'rollorver' ` option parameter to the install, like:

```ts
await newDS.install(params, { migration: 'rollover' })
```

We can also do it later if this needs to get merged before FF

Reply:
Great idea, will add it in the followup PR, Thank you @semd